hubploy/auth.py

```diff
@@ -82,3 +123,24 @@ def cluster_auth_gcloud(deployment, project, cluster, zone, service_key):
         'get-credentials', cluster
     ])
 
+
+def cluster_auth_aws(deployment, project, cluster, zone, service_key):
+    """
+    Setup AWS authentication with service_key
+
+    This changes *global machine state* on what current kubernetes cluster is!
+    """
+    service_key_path = os.path.join(
+        'deployments', deployment, 'secrets', service_key
+    )
+
+    if not os.path.isfile(service_key_path):
+        raise FileNotFoundError(f'The service_key file {service_key_path} does not exist')
+
+    os.environ['AWS_SHARED_CREDENTIALS_FILE'] = os.path.abspath(service_key_path)
+
+    subprocess.check_call([
```

Comment:
You can set the environment just for this call, instead of modifing it for the current process. something like:

```
aws_env = os.environ.copy()
aws_env['NEW_KEY'] = 'SOMETHING'
subprocess.check_call([...], env=aws_env)
```

I might've gotten the method / param names wrong, but doing this would make sure we limit the environment changes to as few places as possible.

Reply:
a context manager would be nice here.

Reply:
added this, but we're getting an authentication error, see next comment